snapcraft/internal/build_providers/_multipass/_multipass_command.py

```diff
@@ -251,6 +249,14 @@ def execute(
 
         return output
 
+    def exists(self, *, instance_name: str) -> bool:
```

Comment:
Why not just make this part of the Multipass class?

Reply:
The current tests make assumptions on usage of `info` as part of its mock strategy.  We could refactor those tests and move it to multipass.  Strictly speaking, I agree it belongs there.